editor/src/clj/editor/model_scene.clj

```diff
@@ -400,14 +433,15 @@
                                     :icon mesh-icon
                                     :view-types [:scene :text]))
 
-(defn- update-vb [^GL2 gl ^VertexBuffer vb data]
-  (let [{:keys [mesh world-transform vertex-space vertex-attribute-bytes]} data
-        world-transform (doto (Matrix4d.) (math/clj->vecmath world-transform))]
-    (mesh->vb! vb world-transform vertex-space vertex-attribute-bytes mesh)))
+(defn- update-vb [^GL2 _gl ^VertexBuffer vb data]
+  (let [{:keys [mesh-renderable-data ^Matrix4d world-transform ^Matrix4d normal-transform vertex-attribute-bytes]} data]
+    (mesh->vb! vb world-transform normal-transform vertex-attribute-bytes mesh-renderable-data)
+    vb))
 
 (defn- make-vb [^GL2 gl data]
-  (let [{:keys [mesh vertex-description]} data
-        vbuf (vtx/make-vertex-buffer vertex-description :dynamic (alength ^ints (:indices mesh)))]
+  (let [{:keys [mesh-renderable-data vertex-description]} data
+        position-data (:position-data mesh-renderable-data)
+        vbuf (vtx/make-vertex-buffer vertex-description :dynamic (count position-data))]
```

Comment:
The `:indices` have been used to construct a non-indexed triangle list already in the  `renderable-mesh-set` output, so at this point we can just count the number of positions.